core/store/src/trie/mod.rs

```diff
@@ -318,19 +318,19 @@ impl RawTrieNode {
                     }
                     pos <<= 1;
                 }
-                cursor.write_u16::<LittleEndian>(bitmap).unwrap();
+                out.extend(bitmap.to_le_bytes());
                 for child in children.iter() {
                     if let Some(hash) = child {
-                        cursor.write_all(hash.as_ref()).unwrap();
+                        out.extend(hash.as_bytes());
                     }
                 }
             }
             // size <= 1 + 4 + key_length + 32
             RawTrieNode::Extension(key, child) => {
-                cursor.write_u8(EXTENSION_NODE).unwrap();
-                cursor.write_u32::<LittleEndian>(key.len() as u32).unwrap();
-                cursor.write_all(key).unwrap();
-                cursor.write_all(child.as_ref()).unwrap();
+                out.push(EXTENSION_NODE);
+                out.extend((key.len() as u32).to_le_bytes());
+                out.extend(key);
+                out.extend(child.as_bytes());
             }
         }
     }
```

Comment:
Since you’re cleaning up dead_code, allow(dead_code) below can be replaced by `#[cfg(test)]`